pkg/admissioncontroller/webhooks/admission/auditpolicy/admission.go

```
@@ -0,0 +1,285 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auditpolicy
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"reflect"
+	"time"
+
+	acadmission "github.com/gardener/gardener/pkg/admissioncontroller/webhooks/admission"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	"github.com/gardener/gardener/pkg/controllermanager/controller/shoot"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+	"github.com/gardener/gardener/pkg/utils/version"
+
+	"github.com/go-logr/logr"
+	admissionv1 "k8s.io/api/admission/v1"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	audit_internal "k8s.io/apiserver/pkg/apis/audit"
+	auditv1 "k8s.io/apiserver/pkg/apis/audit/v1"
+	auditv1alpha1 "k8s.io/apiserver/pkg/apis/audit/v1alpha1"
+	auditv1beta1 "k8s.io/apiserver/pkg/apis/audit/v1beta1"
+	auditvalidation "k8s.io/apiserver/pkg/apis/audit/validation"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const (
+	// HandlerName is the name of this admission webhook handler.
+	HandlerName = "auditpolicy_validator"
+	// WebhookPath is the HTTP handler path for this admission webhook handler.
+	WebhookPath = "/webhooks/validate-audit-policies"
+
+	auditPolicyConfigMapDataKey = "policy"
+)
+
+var (
+	runtimeScheme = runtime.NewScheme()
+	codecs        = serializer.NewCodecFactory(runtimeScheme)
+	decoder       = codecs.UniversalDecoder()
+	_             = auditv1alpha1.AddToScheme(runtimeScheme)
+	_             = auditv1beta1.AddToScheme(runtimeScheme)
+	_             = auditv1.AddToScheme(runtimeScheme)
+	_             = audit_internal.AddToScheme(runtimeScheme)
```

Please don't ignore errors in initializations.
Please use an `init` func instead and `utilruntime.Must`, e.g like this:
https://github.com/gardener/gardener/blob/bcdf52d75e2df05c6ff2ee49524964a8c2b05302/cmd/gardener-admission-controller/app/gardener_admission_controller.go#L63-L71

---

Thanks very helpful!